numpy/lib/tests/test_function_base.py

```diff
@@ -4142,7 +4142,58 @@ def test_closest_observation(self):
         assert_equal(4, np.quantile(arr[0:9], q, method=m))
         assert_equal(5, np.quantile(arr, q, method=m))
 
+    def test_inf_err(self):
 
+        m = "inverted_cdf"
+        q = 0.5
+        arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        wgt = np.ones(10)
+
+        for i in range(len(arr)):
+            wgt[i] = np.inf
+            with pytest.raises(ValueError) as ex:
+                a = np.quantile(arr, q, weights=wgt, method=m)
+            assert "Weights must be non-infinite" in str(ex)
+            wgt[i] = 1
```

Comment:
Probably sufficient to just check a single index rather than all of them. For the message check, I think we usually prefer using the `match` builtin to `pytest.raises` these days.